src/codegen/c.zig

```diff
@@ -386,52 +386,97 @@ pub const DeclGen = struct {
         try dg.renderDeclName(decl, writer);
     }
 
-    fn renderInt128(
+    fn renderBigIntConst(
         writer: anytype,
-        int_val: anytype,
+        val: BigIntConst,
+        is_signed: bool,
     ) error{ OutOfMemory, AnalysisFail }!void {
-        const int_info = @typeInfo(@TypeOf(int_val)).Int;
-        const is_signed = int_info.signedness == .signed;
-        const is_neg = int_val < 0;
-        comptime assert(int_info.bits > 64 and int_info.bits <= 128);
+        const bits = val.bitCountTwosComp();
+        if (bits <= 64) {
+            if (is_signed) {
+                return writer.print("{d}", .{val.to(i64) catch unreachable});
+            } else {
+                return writer.print("{d}u", .{val.to(u64) catch unreachable});
+            }
+        }
 
-        // Clang and GCC don't support 128-bit integer constants but will hopefully unfold them
-        // if we construct one manually.
-        const magnitude = std.math.absCast(int_val);
+        assert(bits <= 128);
+
+        // Clang and GCC don't support 128-bit integer constants but will
+        // hopefully unfold them if we construct one manually.
+        const magnitude = if (val.positive)
+            val.to(u128) catch unreachable
+        else
+            std.math.absCast(val.to(i128) catch unreachable);
 
         const high = @truncate(u64, magnitude >> 64);
         const low = @truncate(u64, magnitude);
 
         // (int128_t)/<->( ( (uint128_t)( val_high << 64 )u ) + (uint128_t)val_low/u )
-        if (is_signed) try writer.writeAll("(int128_t)");
-        if (is_neg) try writer.writeByte('-');
+        if (is_signed) {
+            try writer.writeAll("(int128_t)");
+            if (!val.positive)
+                try writer.writeByte('-');
+        } else {
+            try writer.writeAll("(uint128_t)");
+        }
+
+        try writer.print("(", .{});
 
-        assert(high > 0);
-        try writer.print("(((uint128_t)0x{x}u<<64)", .{high});
+        if (high > 0)
+            try writer.print("((uint128_t)0x{x}u << 64)", .{high});
+
+        if (high > 0 and low > 0)
+            try writer.print(" + ", .{});
 
         if (low > 0)
-            try writer.print("+(uint128_t)0x{x}u", .{low});
+            try writer.print("(uint128_t)0x{x}u", .{low});
 
         return writer.writeByte(')');
     }
 
-    fn renderBigIntConst(
+    fn renderInt128(
```

Comment:
I dont understand why this fn is necessary.

Reply:
The reason I did those changes (renderInt128/renderBigIntConst refactoring/renaming) is not obvious anymore, I'll remove unneeded changes.